### PR TITLE
Optimize loading2

### PR DESCRIPTION
**Pull request by thardeck**
Supersedes #585 
I left out the skip_validation option for the method load! because it would be pointless with it I suppose.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged aea1ad02d1ec31ab719475b9ecb7ccd1c718e126 into b94bcdbf17145fccab16c76b11872884be1fcea4 )

File: lib/system_description.rb [hunk 1]
```diff
@@ -61,10 +61,14 @@ def load!(name, store, options = {})
     # loading of the system description succeeds.
     def load(name, store, options = {})
       manifest = Manifest.load(name, store.manifest_path(name))
-      manifest.validate
+      if !options[:skip_validation]
```
Comment (aduffeck) [inline lib/system_description.rb]: Please use a trailing if clause when the statement is short and there are no other branches.

File: lib/system_description.rb [hunk 2]
```diff
@@ -61,10 +61,14 @@ def load!(name, store, options = {})
...
+        manifest.validate
+      end
 
       description = from_hash(name, store, manifest.to_hash)
-      description.validate_file_data
+      if !options[:skip_validation]
```
Comment (aduffeck) [inline lib/system_description.rb]: Same here.

**Comment by aduffeck** [comment]
LGTM besides the nitpicks.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 2a44132db9d9cb5e5dd51f6a2c474a2825935969 into b94bcdbf17145fccab16c76b11872884be1fcea4 )

**Comment by tgoettlicher** [comment]
lgtm

**Pull request merged**